### DISCUSSION TITLE
Omnet++ dosen't working on new version of Ubuntu

### DISCUSSION THREAD

**Discussion by MSklodowski** [omnetpp/omnetpp#1223] [Q&A]
Hello,

I was using Omnet++ simulator version 6.0.1 on Ubuntu version 20.04.6 LTS with 2 libraries: INET version 4.4.1 and simu5G version 1.2.1. But due to dual boot issue, I updated Ubuntu to 22.04.5 LTS. The system itself started working properly, but the Omnet++ simulator stopped starting and I got the following error:

Simulation terminated with exit code: 7f
Working directory: /home/sqodo/omnetpp-6.0.1/projects/advancetopo
Command line: advancetopo -m -n .:../inet4.5/examples:../inet4.5/showcases:../inet4.5/src:../inet4.5/tests/validation:../inet4.5/tests/networks:../inet4.5/tutorials:../simu5g/emulation:../simu5g/simulations:../simu5g/src -x inet.common.selfdoc;inet.linklayer.configurator.gatescheduling.z3;inet.emulation;inet.showcases.visualizer.osg;inet.examples.emulation;inet.showcases.emulation;inet.transportlayer.tcp_lwip;inet.applications.voipstream;inet.visualizer.osg;inet.examples.voipstream;simu5g.simulations.LTE.cars;simu5g.simulations.NR.cars;simu5g.nodes.cars --image-path=../inet4.5/images:../simu5g/images -l ../inet4.5/src/INET -l ../simu5g/src/simu5g omnetpp.ini

Environment variables:
PATH=/home/sqodo/omnetpp-6.0.1/bin::/home/sqodo/omnetpp-6.0.1/bin:/home/sqodo/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
LD_LIBRARY_PATH=/home/sqodo/omnetpp-6.0.1/lib::/home/sqodo/omnetpp-6.0.1/projects/inet4.5/src:/home/sqodo/omnetpp-6.0.1/projects/simu5g/src:
OMNETPP_RELEASE=omnetpp-6.0.1
OMNETPP_IMAGE_PATH=/home/sqodo/omnetpp-6.0.1/images

What should i do? Should I update the libraries or should I go back to the previous version of Ubuntu? Thank you in advance for your help

**Comment by rhornig** [comment]
Did you re-configure and re-built both OMNeT++ and INET? If the underlying libraries are changed there is a good chance that something will go wrong without rebuilding everything. Any change in the memory layout of any data structure in the libraries can result in crashes.

also: you can also try installing INET + OMNeT++ with `opp_env` which will give you a stable envrionment regardless of the underlying operation system (as it is using NIX as a package manager)

**Comment by MSklodowski** [comment]
Well, I tried it, but it dosen't help. So i decided to re-install my project and libraries. That error which i mentioned disappeared. Right now I got erorr in .ned file with one of Simu5G elements. 
![Zrzut ekranu z 2024-09-25 20-59-38](https://github.com/user-attachments/assets/1299f7ac-a58f-4d6c-a18b-8173d515adc5)

**Reply by rhornig** [reply]
What file is this? Because I see no files in Simu5G that contains the above code.

**Comment by MSklodowski** [comment]
Yeah, I remembered that my friend whose project I am using made modifications to the files in the src folder for the Simu5G library. I'm working on it. Thank you for your help.
